spec/src/main/java/io/a2a/json/JsonUtil.java

```diff
@@ -0,0 +1,729 @@
+     * @see TaskArtifactUpdateEvent
+     */
+    static class StreamingEventKindTypeAdapter extends TypeAdapter<StreamingEventKind> {
+        // Create separate Gson instance without the StreamingEventKind adapter to avoid recursion
+        private final Gson delegateGson = new GsonBuilder()
+                .registerTypeAdapter(OffsetDateTime.class, new OffsetDateTimeTypeAdapter())
+                .registerTypeAdapter(TaskState.class, new TaskStateTypeAdapter())
+                .registerTypeAdapter(Message.Role.class, new RoleTypeAdapter())
+                .registerTypeAdapter(Part.Kind.class, new PartKindTypeAdapter())
+                .registerTypeHierarchyAdapter(Part.class, new PartTypeAdapter())
+                .registerTypeHierarchyAdapter(FileContent.class, new FileContentTypeAdapter())
+                .create();
+
+        @Override
+        public void write(JsonWriter out, StreamingEventKind value) throws java.io.IOException {
+            if (value == null) {
+                out.nullValue();
+                return;
+            }
+            // Delegate to Gson's default serialization for the concrete type
+            if (value instanceof Task task) {
+                delegateGson.toJson(task, Task.class, out);
+            } else if (value instanceof Message message) {
+                delegateGson.toJson(message, Message.class, out);
+            } else if (value instanceof TaskStatusUpdateEvent event) {
+                delegateGson.toJson(event, TaskStatusUpdateEvent.class, out);
+            } else if (value instanceof TaskArtifactUpdateEvent event) {
+                delegateGson.toJson(event, TaskArtifactUpdateEvent.class, out);
+            } else {
+                throw new JsonSyntaxException("Unknown StreamingEventKind implementation: " + value.getClass().getName());
+            }
+        }
+
+        @Override
+        public@Nullable StreamingEventKind read(JsonReader in) throws java.io.IOException {
+            if (in.peek() == com.google.gson.stream.JsonToken.NULL) {
+                in.nextNull();
+                return null;
+            }
+
+            // Read the JSON as a tree so we can inspect the "kind" field
+            com.google.gson.JsonElement jsonElement = com.google.gson.JsonParser.parseReader(in);
+            if (!jsonElement.isJsonObject()) {
+                throw new JsonSyntaxException("StreamingEventKind must be a JSON object");
+            }
+
+            com.google.gson.JsonObject jsonObject = jsonElement.getAsJsonObject();
+            com.google.gson.JsonElement kindElement = jsonObject.get("kind");
+            if (kindElement == null || !kindElement.isJsonPrimitive()) {
+                throw new JsonSyntaxException("StreamingEventKind must have a 'kind' field");
+            }
+
+            String kind = kindElement.getAsString();
+            // Use the delegate Gson to deserialize to the concrete type
+            return switch (kind) {
```

Comment:
code does not reflect the current 1.0 spec that no longer has a `kind`
and the field names are in camelCase now: https://a2a-protocol.org/latest/specification/#433-push-notification-payload